extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/security/CORS.java

```diff
@@ -0,0 +1,222 @@
+package io.quarkus.vertx.http.security;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import io.quarkus.vertx.http.runtime.cors.CORSConfig;
+import io.quarkus.vertx.http.runtime.security.HttpSecurityUtils;
+import io.smallrye.common.annotation.Experimental;
+
+/**
+ * This class provides a way to configure the Quarkus cross-origin resource sharing (CORS) filter at the HTTP layer level.
+ */
+@Experimental("This API is currently experimental and might get changed")
+public sealed interface CORS permits CORS.Builder.CORSImpl {
+
+    /**
+     * Creates the CORS configuration builder.
+     *
+     * @return new {@link Builder} instance
+     */
+    static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Create a new CORS configuration with given origins.
+     *
+     * @param origins see {@link Builder#origins(Set)}
+     * @return CORS instance, which should be passed to the {@link HttpSecurity} event
+     */
+    static CORS origins(String... origins) {
```

Comment:
@michalvavrik IMHO we should retain your original `origin(String)`, with `origins(Set<String>)` covering more advanced cases

Reply:
I don't really mind anyway, I'll follow your suggestion.